src/Mado/QueryBundle/Component/Meta/GraphWalker.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace Mado\QueryBundle\Component\Meta;
+
+/**
+ * @since Class available since Release 2.1.1
+ */
+interface GraphWalker
+{
+    public function buildPathBetween($start, $end) : bool;
```

Comment:
can you add type of start and end?

Reply:
Nope. We can improve names. $firstNode and $lastNode could be better understandable. $start, in the graph, is the node where the path start. $end is the final node.

 - [ ] firstNode, lastNode
 - [ ] initialNode, finalNode
 - [ ] try again

select one of previous choice ^_^

Reply:
We cant because we dont know what a node could be. A string maybe. Actually we use Fully Qualified Class Name or relation name. In this context start and end are simply node.

Reply:
ok